pkg/service/backup/restore_worker_data.go

```diff
@@ -25,22 +25,30 @@ func (w *restoreWorker) restoreData(ctx context.Context, run *RestoreRun, target
 		w.hosts = nil
 
 		manifestHandler := func(miwc ManifestInfoWithContent) error {
-			w.miwc = miwc
-			w.Logger.Info(ctx, "Restoring manifest", "manifest", miwc.ManifestInfo)
 			// Check if manifest has already been processed in previous run
 			if !w.resumed && run.ManifestPath != miwc.Path() {
 				return nil
 			}
+
+			w.Logger.Info(ctx, "Restoring manifest", "manifest", miwc.ManifestInfo)
+			defer w.Logger.Info(ctx, "Restoring manifest finished", "manifest", miwc.ManifestInfo)
+
+			w.miwc = miwc
 			run.ManifestPath = miwc.Path()
 
 			return miwc.ForEachIndexIterWithError(target.Keyspace, w.restoreFiles(ctx, run, target))
 		}
 
 		if err := w.forEachRestoredManifest(ctx, w.location, manifestHandler); err != nil {
+			w.Logger.Info(ctx, "Restoring location finished", "location", w.location)
```

Comment:
Why to log this Info message on error ?
It may be misleading. Error will be logged at the top of invocation tree anyway.

Reply:
I wrote that in this way so that it's consistent with logging information about table/manifest restore finished. In those two cases, I am using defer without any error checking, so it logs the information regardless of whether the restore finished with success or error (that's also the reason why I decided to use word "finished", so that it does not implicate success). 
I could add some error checking to those defers which would log either restore location/manifest/table completed/failed
and apply the same logic here. What do you think about that?

Reply:
Thanks for the explanation, it's fine.